src/wp-includes/theme.php

```diff
@@ -4093,3 +4093,21 @@ function create_initial_theme_features() {
 function wp_is_block_theme() {
 	return wp_get_theme()->is_block_theme();
 }
+
+/**
+ * Adds default theme supports for block themes when the 'setup_theme' action
+ * fires.
+ *
+ * @since 5.9.0
+ * @access private
+ */
+function _add_default_theme_supports() {
+	if ( wp_is_block_theme() ) {
+		add_theme_support( 'post-thumbnails' );
+		add_theme_support( 'responsive-embeds' );
+		add_theme_support( 'editor-styles' );
+		add_theme_support( 'html5', array( 'style', 'script' ) );
```

Comment:
I think it would make sense to extend the HTML5 support for other elements too. At least for components which are used by blocks like `comment-form`/`comment-list`?

Reply:
What do you think @youknowriad? I'm just copying over what's already in the plugin 😀

Reply:
I also just copied what the themers suggested from 2022 theme. that said IMO, everything should be supported. Can we do `*` or something like that?

Reply:
> that said IMO, everything should be supported. Can we do * or something like that?

No, `add_theme_support( 'html5', ... )` intentionally requires an array of specific types to check, see [this comment](https://core.trac.wordpress.org/ticket/24932#comment:4) for context, and some more history in the description of https://core.trac.wordpress.org/ticket/51657.

Reply:
I'm not sure I understand the reasoning there though?

Reply:
There’s no guarantee that a current theme can support a new entry added  in a future WordPress Version without having to adjust something first. That’s why the theme supports are opt-in.

Reply:
I think that's probably not true for block themes as each block is independent and not controlled by the theme.

Reply:
What should we do in the short term for 5.9?

Reply:
We can land this as is or by adding as much elements as available to at least be on the same page as the Gutenberg plugin.

Reply:
As mentioned in https://github.com/WordPress/wordpress-develop/pull/2029#discussion_r765217669, we only have to add the two for the comments blocks.

```suggestion
		add_theme_support( 'html5', array( 'comment-form', 'comment-list', 'style', 'script' ) );
```

 `search-form`, `gallery`, `caption`, and `navigation-widgets` (not 100% sure) seem to be irrelevant for block themes.

Reply:
Indeed, declaring the support of `navigation-widgets` currently doesn't change anything to the markup of the navigation widgets (page/category list, etc). 

It's supposed to add a `<nav>` container to the widget. This should probably be reported upstream since it's a regression from Classic themes. I'll open an issue on Gutenberg for this.

Reply:
@ocean90 can you add those 2 https://github.com/WordPress/wordpress-develop/pull/2029/files#r766453524 upstream in Gutenberg please? Will go ahead with the suggestion here. But needs to be synched with Gutenberg too.